### PR TITLE
docs([DST-586]): remove examples from checkbox page and add guidline section

### PR DESCRIPTION
**Pull request by sebald**
# Description

- Removed the examples section
- added guidline about the difference between a checkbox and a switch
- site clean up

# Reviewers:

@marigold-ui/developer
@marigold-ui/designer

# Issue

Ref [DST-586]

[DST-586]: https://reservix.atlassian.net/browse/DST-586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sarahgm** [comment]
The Title should contain the issue number :)

File: docs/content/components/form/checkbox/checkbox.mdx [hunk 1]
```diff
@@ -36,15 +30,17 @@ The checkbox component is comprised of a checkbox label and a checkbox input. If
 
 ## Usage
 
+Checkboxes allow user to to toggle a single option or to select multiple options from a list. Each checkbox works independently, so selecting one does not affect the others. Users can choose none, one, or many items.
```
Comment (sarahgm) [inline docs/content/components/form/checkbox/checkbox.mdx:33]: The following section also started with "Checkboxes" maybe you find another start?

File: docs/content/components/form/checkbox/checkbox.mdx [hunk 2]
```diff
@@ -53,12 +49,12 @@ If you have number of options is moderate to large, typically more than 10-15 op
...
         unoptimized
-        src="/checkbox/checkbox-dont-select.jpg"
-        alt="Don't use checkboxes when only one item can be selected."
+        src="/checkbox/checkbox-dont-long-options.jpg"
+        alt="Don't use checkbox with long list of options."
       />
     </Dont.Figure>
     <Dont.Description>
-      Don't use checkboxes when only one item can be selected.
+      Don't use checkbox with long list of options.
```
Comment (sarahgm) [inline docs/content/components/form/checkbox/checkbox.mdx:57]: how long is long list of options?

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: docs/content/components/form/checkbox/checkbox.mdx (hunk 2 above)
Comment (sebald) [reply docs/content/components/form/checkbox/checkbox.mdx:57]: I just switched places. Same text as before :P

Comment (sarahgm) [reply docs/content/components/form/checkbox/checkbox.mdx:57]: but you also revising it now or not? Why not correct it?

Comment (sebald) [reply docs/content/components/form/checkbox/checkbox.mdx]: jup updated it

**Review by sarahgm — APPROVED** [review]

**Pull request merged**